### PR TITLE
Added proxy server argument to InfluxClient ctor. Used Base64 authent…

### PR DESCRIPTION
**Pull request by hadamarda**
1. Added argument to InfluxDbClient ctor. that may specify proxy server and credentials.
2. Use Base64 authentication

### CODE REVIEW

**Comment by mvadu** [comment]
Unfortunately most of the checks are failing. Can you please check once from your end. I will take a look later this week on the remaining failures.

File: src/DataStructures/InfluxDBClient.cs [hunk 1]
```diff
@@ -238,19 +240,36 @@ public int Port
...
-           //if (!(String.IsNullOrWhiteSpace(InfluxDBUserName) && String.IsNullOrWhiteSpace(InfluxDBPassword)))
-           //    _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
-           //        Convert.ToBase64String(Encoding.GetEncoding("iso-8859-1").GetBytes($"{InfluxDBUserName}:{InfluxDBPassword}")));
-            _client.DefaultRequestHeaders.ConnectionClose = false;
+        public InfluxDBClient(string InfluxUrl, string UserName, string Password) :
+            this(InfluxUrl, UserName, Password, null)
+        {            
         }
 
+        public InfluxDBClient(string InfluxUrl, string UserName, string Password, HttpClientHandler httpClientHandler)
```
Comment (mvadu) [inline src/DataStructures/InfluxDBClient.cs]: Please add comments to document the httpClientHandler's role

File: src/DataStructures/InfluxDBClient.cs [hunk 2]
```diff
@@ -238,19 +240,36 @@ public int Port
...
+                this._influxDBPassword = Password;                
+
+                if (httpClientHandler != null)                               
+                    _client = new HttpClient(httpClientHandler, true);                
+                else
+                    _client = new HttpClient();
+
+                if (!(String.IsNullOrWhiteSpace(InfluxDBUserName) && String.IsNullOrWhiteSpace(InfluxDBPassword)))
+                    _client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue("Basic",
+                        Convert.ToBase64String(Encoding.GetEncoding("iso-8859-1").GetBytes($"{InfluxDBUserName}:{InfluxDBPassword}")));
```
Comment (mvadu) [inline src/DataStructures/InfluxDBClient.cs]: this was commented to fix #38. Have you tested to confirm it works with non ascii characters in password? Correct way is to use UTF-8, but I am not sure if Influx supports UTF-8 Authorization header.

**Comment by hadamarda** [comment]
As far as I understand, the problem is in the unicode password.
When using some simple password, like 'test1234', the tests pass fine.

**Comment by mvadu** [comment]
I know.. I commented those lines for a reason (to fix #38). Please comment out the base 64 part, and tests will pass.

**Pull request merged**